src/Crypto.ts

```diff
@@ -1,150 +1,152 @@
-import * as crypto from 'crypto'
+import { CryptoInterface, Cipher, Decipher, AlgorithmGCM, cryptoConstants, deconstructChunk } from './util'
 import { spawn, Worker, ModuleThread } from 'threads'
-import { Buffer } from 'buffer/'
 
 export default class Crypto {
-  private algorithm: string
-  private initVector: Buffer
-  private key: Buffer
-  private cipher: crypto.Cipher
-  private decipher: crypto.Decipher
+  private masterKey: Buffer
+  private algorithm: AlgorithmGCM
+  // Web workers
   private useWebWorkers: boolean
-  private cryptorWorker?: ModuleThread
+  private cryptoWorker?: ModuleThread
+  // Sync members
+  private cipher: Cipher
+  private decipher: Decipher
+  // Crypto lib
+  private cryptoLib: CryptoInterface
```

Comment:
We can make it general later (for the web workers) down the line by actually providing an interface and a byo crypto worker but yeah for now this can stay like this